docs/guide/config.md

```diff
@@ -0,0 +1,157 @@
+
+Configuration takes a few forms:
+
+- **Global config:** control over basic operation of `miniscope-io` like logging,
+  location of user directories, plugins, etc.
+- **Device config:** control over the operation of specific devices and miniscopes like
+  firmware versions, ports, capture parameters, etc.
+- **Runtime/experiment config:** control over how a device behaves when it runs, like
+  plotting, data output, etc.
+
+## Global Config
+
+Global config uses the {class}`~miniscope_io.models.config.Config` class
+
+Config values can be set (in order of priority from high to low, where higher
+priorities override lower priorities)
+
+* in the arguments passed to the class constructor (not user configurable)
+* in environment variables like `export MINISCOPE_IO_LOG_DIR=~/`
+* in a `.env` file in the working directory
+* in a `mio_config.yaml` file in the working directory
+* in the `tool.miniscope_io.config` table in a `pyproject.toml` file in the working directory
+* in a user `mio_config.yaml` file in the user directory (see [below](user-directory))
+* in the global `mio_config.yaml` file in the platform-specific data directory
+  (use `mio config global path` to find its location)
+* the default values in the {class}`~miniscope_io.models.config.Config` model
+
+Parent directories are _not_ checked - `.env` files, `mio_config.yaml`, and `pyproject.toml`
+files need to be in the current working directory to be discovered.
+
+You can see your current configuration with `mio config`
+
+(user-directory)=
+### User Directory
+
+The configuration system allows project-specific configs per-directory with
+`mio_config.yaml` files in the working directory, as well as global configuration
+via `mio_config.yaml` in the system-specific config directory 
+(via [platformdirs](https://pypi.org/project/platformdirs/)). 
+By default, `miniscope-io` does not create new directories in the user's home directory
+to be polite, but the site config directory might be inconvenient or hard to reach,
+so it's possible to create a user directory in a custom location.
+
+`miniscope_io` discovers this directory from the `user_dir` setting from 
+any of the available sources, though the global `mio_config.yaml` file is the most reliable.
+
+To create a user directory, use the `mio config user create` command.
+(ignore the `--dry-run` flag, which are just used to avoid
+overwriting configs while rendering the docs ;)
+
+```{command-output} mio config user create ~/my_new_directory --dry-run
+```   
+
+You can confirm that this will be where miniscope_io discovers the user directory like
+
+```{command-output} mio config user path
+```
+
+If a directory is not supplied, the default `~/.config/miniscope_io` is used:
+
+```{command-output} mio config user create --dry-run
+```
+
+### Setting Values
+
+```{todo}
+Implement setting values from CLI.
+
+For now, please edit the configuration files directly.
```

Comment:
It is also related to my comments in #76, but I'm not sure which is easier between these two (though it wouldn't hurt to have these two work nicely). I feel that using text editors could be easier than using CLI for our target users (~undefined), but I am unsure.

Reply:
totally ya. after seeing you working with the cli setting config values i think that would be really nice to have, and that should also edit the relevant files, so they should be made to be equivalent imo if we do set config values from the cli - i.e. editing from cli should change the value in the config files as if one were to edit it with a text editor